org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleBuildSupport.java

```diff
@@ -54,9 +58,21 @@ public void update(IProject project, boolean force, IProgressMonitor monitor) th
 
 	@Override
 	public boolean isBuildFile(IResource resource) {
-		return resource != null
-				&& resource.getType()== IResource.FILE
-				&& resource.getName().endsWith(GRADLE_SUFFIX);
+		if (resource != null && resource.getType() == IResource.FILE && resource.getName().endsWith(GRADLE_SUFFIX) && resource.getProject() != null && ProjectUtils.isGradleProject(resource.getProject())) {
+			try {
+				IJavaProject javaProject = JavaCore.create(resource.getProject());
+				IPath outputLocation = javaProject.getOutputLocation();
```

Comment:
return outputLocation == null || !outputLocation.isPrefixOf(resource.getFullPath())

Reply:
that's nitpicking ;-)

however, please fix the commit message, it's non-sensical

Reply:
Fixed.